File: src/my_project/easy_problems/from251to300/backspace_string_compare.py
from typing import List

class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        
        # Helper to process a string
        def process(str):
          newS = ""
          for i in str:
            # Backspace to drop last character
            if i == "#":
              newS = newS[:-1]
            # Normal character so add it
            else:
              newS += i
          return newS
        
        # Run the helper for both strings and compare
        if process(s) == process(t) : 
          return True
        return False
    
solution = Solution()

print(solution.backspaceCompare(s = "ab#c", t = "ad#c"))

print(solution.backspaceCompare(s = "ab##", t = "c#d#"))

print(solution.backspaceCompare(s = "a#c", t = "b"))

sample = 'abcd'
print(sample[:-1])

File: my_project_typescript/easy_problems_typescript/from151to200/backspaceStringCompare.ts
function backspaceCompare(s: string, t: string): boolean {

    function returnString(s: string) : string{
        let newString: string = ''
        
        for (let letter of s){
            if (letter === '#'){
                newString = newString.slice(0,newString.length -1)
            }
            else{
                newString += letter
            }
        }
        return newString
    }

    

    return returnString(s) === returnString(t)

};

let sample = 'abcd'
console.log(sample.slice(0,sample.length-1))